### ISSUE TITLE
CMake HDF5 non-MPI

### ISSUE THREAD

**Issue by scemama** [QMCPACK/qmcpack#2301]
**Describe the bug**
On a Debian 9 machine, I installed HDF5 using apt install hdf5-dev.
The hdf5.h file is located in /usr/include/hdf5/serial/
I ran cmake 3.7.2 with -DQMC_MPI=0 and everything went fine.
When I compiled, I got this message:
```
In file included from /home/scemama/qmcpack/src/io/hdf_archive.h(18),
                 from /home/scemama/qmcpack/src/QMCWaveFunctions/SPOSet.h(27),
                 from /home/scemama/qmcpack/src/QMCWaveFunctions/SPOSetBuilder.h(27),
                 from /home/scemama/qmcpack/src/QMCWaveFunctions/SPOSetBuilderFactory.h(20),
                 from /home/scemama/qmcpack/src/QMCWaveFunctions/WaveFunctionFactory.h(24),
                 from /home/scemama/qmcpack/src/QMCWaveFunctions/WaveFunctionPool.h(22),
                 from /home/scemama/qmcpack/src/QMCWaveFunctions/WaveFunctionPool.cpp(19):
/home/scemama/qmcpack/src/io/hdf_datatype.h(17): catastrophic error: cannot open source file "hdf5.h"
  #include <hdf5.h>
```
To make it compile properly, I had to 
```export CPLUS_INCLUDE_PATH=${CPLUS_INCLUDE_PATH}:/usr/include/hdf5/serial```

**To Reproduce**
1. release version or git commit hash being built
commit 7e7a6869aa611b062a71336a9afa53712e174ed2
Merge: ef8b21ced c63c50cb9
Author: Ye Luo <yeluo@anl.gov>
Date:   Sun Feb 9 16:03:25 2020 +0800

    Merge pull request #2294 from mmorale3/fix_lindep_oao
    
    Fix linear dependency issue

2. cmake command
cmake -DQMC_MPI=0 ..

3. full program/test invocation command
N/A
4. additional steps
Install linbhdf5-dev with package manager on Debian9.

**Expected behavior**

**System:**
 - system name [e.g. fusiont5, summit] lpqlx139 (my desktop computer!)
 - modules loaded [e.g. output of `module list`] 
 - other systems where this is reproducible [e.g. "my laptop", "none"]

**Additional context**

Thank you!

**Comment by markdewing** [comment]
It could be an issue with the version of CMake.  Try a newer version and sees if the problem reproduces. (unfortunately, I don't know precisely which version you should try)

**Comment by markdewing** [comment]
I can reproduce the issue on Ubuntu 18.04 with cmake 3.7.2 (and 3.6.1).  It works with 3.8.1 (and presumably newer.) I didn't test 3.8.0.

**Comment by scemama** [comment]
Thank you for your help.
OK, so then you should just change the CMake requirement and set it to 3.8.1, right?

**Comment by prckent** [comment]
Thanks Anthony. It seems that indeed we need to raise the minimum version.

The oldest version that we intentionally test with is cmake 3.10.2, released 2018-01-18. As far as I can recall, we have never found a reason to not use the latest version of cmake. We are going to have to require more recent versions before long due to the improved CUDA language support.

**Comment by prckent** [comment]
See also https://github.com/QMCPACK/qmcpack/pull/2289 which requires 3.10.  Some variant of this PR will be merged in after v3.9.0 is out, which should be tomorrow. Sorry for the trouble in the meantime.

**Comment by prckent** [comment]
The CMake requirement on develop is now 3.10. We can reopen this if needed.
